06-TimeManagement/module/module_06.c

```diff
@@ -0,0 +1,79 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+/*
+ *  module_06.c - Test module
+ *
+ *  Copyright (C) 2019 Andrey Pahomov <pahomov.and@gmail.com>
+ */
+#include <linux/init.h>
+#include <linux/module.h>
+#include <linux/kernel.h>
+#include <linux/proc_fs.h>
+#include <linux/uaccess.h>
+#include <linux/device.h>
+
+static int absolute_time;
+static int relation_time;
+
+static struct kobject *kobject_module_06;
+
+static ssize_t time_show(struct kobject *kobj, struct kobj_attribute *attr,
+		      char *buf)
+{
+	return sprintf(buf, "test: %d\n", 123);
+}
+
+static struct kobj_attribute absolute_time_attribute =
+	__ATTR(absolute_time, 0444, time_show, NULL);
+	// __ATTR(bar, 0666, absolute_time_show, absolute_time_store);
+
+static struct kobj_attribute relation_time_attribute =
+	__ATTR(relation_time, 0444, time_show, NULL);
+	// __ATTR(bar, 0666, relation_time_show, relation_time_store);
+
+static struct attribute *attrs[] = {
+	&relation_time_attribute.attr,
+	&absolute_time_attribute.attr,
+	NULL,	/* need to NULL terminate the list of attributes */
+};
+
+static struct attribute_group attr_group = {
+	.attrs = attrs,
+};
+
+
+static int module_06(void)
+{
+	int retval;
+
+	pr_info("%s: module starting\n",  __func__);
+
+	// /sys/kernel/module_06/
+	// module_06_kobj = kobject_create_and_add("module_06", kernel_kobj);
+
+	// /sys/module_06/
+	 kobject_module_06 = kobject_create_and_add("module_06", NULL);
+	if (!kobject_module_06)
+		return -ENOMEM;
+
+	retval = sysfs_create_group(kobject_module_06, &attr_group);
+	if (retval)
+		kobject_put(kobject_module_06);
+
+	pr_info("%s: module start: %d\n",  __func__, retval);
+	return retval;
+}
+
+static void module_06_exit(void)
+{
+	pr_info("%s: module exit\n",  __func__);
+	kobject_put(kobject_module_06);
```

Comment:
Is this call of kobject_put() correct?

Reply:
It works correctly but you're right I need to add sysfs_remove_group

Reply:
What does is mean 'work correctly'? Did not generate segfault.